### PR TITLE
Isolate msbuild from cli dependencies

### LINKED ISSUE #4214: move msbuild install into a subfolder in CLI
## Steps to reproduce
1) Install dotnet CLI from https://dotnetcli.blob.core.windows.net/dotnet/Sdk/1.0.0-preview3-003618/dotnet-dev-win-x64.1.0.0-preview3-003618.zip . By installing, i mean extract the downloaded zip into a directory.

2) Run dotnet.exe restore on the attached project
3) Run dotnet.exe build3 to build the project.
4) Run dotnet.exe pack3 to pack - this will trigger a missing method exception.

This is because msbuild.exe loads up the wrong nuget.commands.dll .

[NetCoreCsproj.zip](https://github.com/dotnet/cli/files/483810/NetCoreCsproj.zip)

## Expected  behavior
it should load the nuget.commands.dll from %(UserProfile).nuget\packages\nuget.build.tasks.pack\3.6.0-beta.1.msbuild.10\build\netstandard1.3\nuget.commands.dll

## Actual behavior
instead it loads the dll from dotnet-dev-win-x64.1.0.0-preview3-003618\sdk\1.0.0-preview3-003618

This is not critical for preview 5 oob as long as nuget pack3 can work by putting the same dependencies in the sdk and cli.

### PR DESCRIPTION
**Pull request by jonsequitur**
This isolates the MSBuild, MSBuildExtensions, and Roslyn file locations into separate folders.

It fixes #4214.

### CODE REVIEW

**Comment by jonsequitur** [comment]
@piotrpMSFT @livarcocc

**Comment by jonsequitur** [comment]
@dotnet-bot test Windows_NT x64 Release Build

File: build/Microsoft.DotNet.Cli.Compile.targets [hunk 1]
```diff
@@ -98,26 +108,36 @@
 
     <!-- Workaround for https://github.com/dotnet/sdk/issues/115 -->
     <ItemGroup>
-      <HackFilesToCopy Include="$(NuGetPackagesDir)\microsoft.build.runtime\$(CLI_MSBuild_Version)\contentFiles\any\netcoreapp1.0\**;$(NuGetPackagesDir)\microsoft.codeanalysis.build.tasks\2.0.0-beta6-60922-08\contentFiles\any\any\**;$(NuGetPackagesDir)\microsoft.testplatform.cli\$(CLI_TestPlatform_Version)\contentFiles\any\any\**" />
+      <HackFilesToCopy Include="$(NuGetPackagesDir)\microsoft.testplatform.cli\15.0.0-preview-20161123-03\contentFiles\any\any\**" />
```
Comment (jgoshi) [inline build/Microsoft.DotNet.Cli.Compile.targets]: Issue 115 looks like it's been fixed? Could you try to remove this workaround (just delete the whole ItemGroup and anything that uses it) and see if the build works and tests pass?

Comment (jonsequitur) [inline build/Microsoft.DotNet.Cli.Compile.targets]: This is still needed.

**Comment by jonsequitur** [comment]
@dotnet-bot test OSX x64 Release Build

File: test/Microsoft.DotNet.Cli.Utils.Tests/GivenAProjectToolsCommandResolver.cs [hunk 2]
```diff
@@ -311,9 +311,7 @@ public void It_does_not_add_fx_version_as_a_param_when_the_tool_does_not_have_th
 
         private ProjectToolsCommandResolver SetupProjectToolsCommandResolver()
         {
-            Environment.SetEnvironmentVariable(
-                Constants.MSBUILD_EXE_PATH,
-                Path.Combine(new RepoDirectoriesProvider().Stage2Sdk, "MSBuild.dll"));
+
```
Comment (krwq) [inline test/Microsoft.DotNet.Cli.Utils.Tests/GivenAProjectToolsCommandResolver.cs]: nit: redundant spaces

**Pull request closed without merging**